sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateSchemaCompatibilityChecker.scala

```diff
@@ -186,6 +186,11 @@ class StateSchemaCompatibilityChecker(
           check(existingStateSchema, newSchema, ignoreValueSchema)
         }
       }
+      // if we have reached this block, the schema has changed in an acceptable way and
```

Comment:
Can we move this out from this PR? This isn't related to purge and warrant a separate JIRA ticket to have better tracking.

Reply:
Hm, but I do think that this PR is related to the cadence of how schema files are written. This is a small change that I do think is somewhat related...